_posts/pl/2023-11-17-czy-wiesz-jak-nalezy-trzymac-powiazane-dane-w-jednym-dokumencie-w-bazie-mongodb.md

```diff
@@ -0,0 +1,141 @@
+---
+layout:    post
+title:     "Czy wiesz jak należy trzymać powiązane dane w jednym dokumencie w bazie MongoDB?"
+date:      2023-11-17T12:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: bmitan
+image: /assets/img/posts/2023-11-17-czy-wiesz-jak-nalezy-trzymac-powiazane-dane-w-jednym-dokumencie-w-bazie-mongodb/data.webp
+tags:
+- mongodb
+- subset
+- extended reference
+- bazy danych
+---
+
+Ważna zasada przy projektowaniu schematu w MongoDB mówi, żeby dane, do których dostęp jest zazwyczaj wspólny, przechowywać razem. Jest to całkowicie sprzeczne z koncepcją normalizacji, podzieleniem danych tak, żeby uniknąć duplikowania i ograniczyć potrzebne miejsce na dane, czyli ze wszystkim tym, co znamy z relacyjnych baz danych.
```

Comment:
Ważna zasada przy projektowaniu schematu w MongoDB mówi, żeby dane, do których dostęp jest zazwyczaj wspólny, przechowywać razem. Jest to całkowicie sprzeczne z koncepcją normalizacji – dzieleniem danych tak, żeby uniknąć duplikowania i ograniczyć potrzebne miejsce na dane – czyli ze wszystkim, co znamy z relacyjnych baz danych.